flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -0,0 +1,366 @@
+		final String taskManagerPrefix = CLUSTER_ID + "-taskmanager-1-";
+		assertEquals(taskManagerPrefix + 1, pod.getMetadata().getName());
+
+		resourceManager.onAdded(Collections.singletonList(new KubernetesPod(flinkConfig, pod)));
+
+		// General modification event
+		resourceManager.onModified(Collections.singletonList(new KubernetesPod(flinkConfig, pod)));
+		assertEquals(1, client.pods().list().getItems().size());
+		assertEquals(taskManagerPrefix + 1, client.pods().list().getItems().get(0).getMetadata().getName());
+
+		// Terminate the pod.
+		pod.setStatus(new PodStatusBuilder()
+			.withContainerStatuses(new ContainerStatusBuilder().withState(
+				new ContainerStateBuilder().withNewTerminated().endTerminated().build())
+				.build())
+			.build());
+		resourceManager.onModified(Collections.singletonList(new KubernetesPod(flinkConfig, pod)));
+
+		// Old pod should be deleted and a new task manager should be created
+		assertEquals(1, client.pods().list().getItems().size());
+		assertEquals(taskManagerPrefix + 2, client.pods().list().getItems().get(0).getMetadata().getName());
+
+		resourceManager.getRpcService().stopService().get();
+	}
+
+	@Test
+	public void testGetWorkerNodesFromPreviousAttempts() throws Exception {
+		// Prepare pod of previous attempt
+		final FlinkKubeClient flinkKubeClient = getFabric8FlinkKubeClient();
+		final String previewPodName = CLUSTER_ID + "-taskmanager-1-1";
+		flinkKubeClient.createTaskManagerPod(new TaskManagerPodParameter(
+			previewPodName,
+			new ArrayList<>(),
+			1024,
+			1,
+			new HashMap<>()));
+		final KubernetesClient client = getKubeClient();
+		assertEquals(1, client.pods().list().getItems().size());
+
+		final String taskManagerPrefix = CLUSTER_ID + "-taskmanager-";
+		final TestingKubernetesResourceManager resourceManager = this.createAndStartResourceManager(flinkConfig);
+		// Register the previous taskmanager, no new pod should be created
+		registerTaskExecutor(resourceManager, new ResourceID(previewPodName));
+		registerSlotRequest(resourceManager);
+		assertEquals(1, client.pods().list().getItems().size());
+
+		// Register a new slot request, a new taskmanger pod will be created with attempt2
+		registerSlotRequest(resourceManager);
+		assertEquals(2, client.pods().list().getItems().size());
+		assertThat(client.pods().list().getItems().stream()
+				.map(e -> e.getMetadata().getName())
+				.collect(Collectors.toList()),
+			Matchers.containsInAnyOrder(taskManagerPrefix + "1-1", taskManagerPrefix + "2-1"));
+
+		resourceManager.getRpcService().stopService().get();
+	}
+
+	private TestingKubernetesResourceManager createAndStartResourceManager(Configuration configuration) throws Exception {
+
+		final TestingRpcService rpcService = new TestingRpcService(configuration);
+		final TestingFatalErrorHandler testingFatalErrorHandler = new TestingFatalErrorHandler();
+		org.apache.flink.runtime.resourcemanager.utils.MockResourceManagerRuntimeServices rmServices =
```

Comment:
Reference this package in `import` then we can avoid long line here.

Reply:
Will fix it.